application-admintools-default/src/main/java/com/xwiki/admintools/internal/data/identifiers/CurrentServer.java

```diff
@@ -0,0 +1,127 @@
+import org.xwiki.component.phase.Initializable;
+import org.xwiki.component.phase.InitializationException;
+
+import com.xwiki.admintools.ServerIdentifier;
+import com.xwiki.admintools.configuration.AdminToolsConfiguration;
+
+/**
+ * Manages the server identifiers and offers endpoints to retrieve info about their paths.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Component(roles = CurrentServer.class)
+@Singleton
+public class CurrentServer implements Initializable
+{
+    @Inject
+    private Provider<List<ServerIdentifier>> supportedServers;
+
+    private ServerIdentifier usedServer;
+
+    @Inject
+    @Named("default")
+    private AdminToolsConfiguration adminToolsConfig;
+
+    /**
+     * Method called by the Component Manager when the component is created for the first time (i.e. when it's looked up
+     * for the first time or if the component is specified as being loaded on startup). If the component instantiation
+     * strategy is singleton then this method is called only once during the lifecycle of the Component Manager.
+     * Otherwise the component is created at each lookup and thus this method is called at each lookup too.
+     *
+     * @throws InitializationException if an error happens during a component's initialization
+     */
+    @Override
+    public void initialize() throws InitializationException
+    {
+        String providedConfigServerPath = adminToolsConfig.getServerPath();
+        findServer(providedConfigServerPath);
+    }
+
+    /**
+     * Retrieves the XWiki configuration file path for the installation.
+     *
+     * @return a {@link String} representing the path to the XWiki configuration file.
+     */
+    public String retrieveXwikiCfgPath()
+    {
+        return usedServer.getXwikiCfgFolderPath();
+    }
+
+    /**
+     * Retrieves the server configuration file path for the installation.
+     *
+     * @return a {@link String} representing the path to the server configuration file.
+     */
+    public String retrieveServerCfgPath()
+    {
+        return usedServer.getServerCfgPath();
+    }
+
+    /**
+     * Verifies if a server type was found and updates the paths.
+     */
+    public void findPaths()
+    {
+        if (usedServer == null) {
+            String providedConfigServerPath = adminToolsConfig.getServerPath();
+            findServer(providedConfigServerPath);
+        }
+    }
+
+    /**
+     * Calls the used server function to retrieve the server identifiers.
+     *
+     * @return a {@link Map} with the info used to identify the server.
+     */
+    public Map<String, String> getServerIdentifiers()
+    {
+        return usedServer.getServerIdentifiers();
+    }
+
+    /**
+     * Go through all supported servers and return the one that is used.
+     *
+     * @param providedConfigServerPath {@link String} server path provided by XWiki configurations.
+     */
+    private void findServer(String providedConfigServerPath)
```

Comment:
I could change the name to identifyUsedServer(config). What do you think?

Regarding what happens if the server identifier is not found, at the moment there is no error sent from the back-end. When I wrote this part, my idea was to verify this directly in the front-end. I'm thinking I could throw an error in the back-end in which I could mention that no server was found, prompt the user to check the configuration file and mention the supported servers?

Reply:
`identify` is still on the `get` side of the problem, like I would expect the method to return the current server, not to change the value of `this.currentServer`

It's ok to do the check on the client side and display a custom message there, but you need to make sure that the server side code handles the case when you don't have an identifier defined. I think  that now it will just throw a NPE somewhere